From: Raj, Deepu <deepu.raj@ti.com>
Date: Tue, 8 May 2012 08:34:47 +0000 (+0530)
Subject: TI814X: Fixed USB PLL divider value for PG3.0
X-Git-Url: http://arago-project.org/git/projects/?p=u-boot-omap3.git;a=commitdiff_plain;h=cbd0e59570e313405f73d8ae4b1f016a1980b308

TI814X: Fixed USB PLL divider value for PG3.0

This patch fixed the divider values used for
UART in the USB PLL config.The divider selection
is done based on the runtime PG value.

Signed-off-by: Raj, Deepu <deepu.raj@ti.com>
---

diff --git a/arch/arm/cpu/arm_cortexa8/ti81xx/sys_info.c b/arch/arm/cpu/arm_cortexa8/ti81xx/sys_info.c
index 141e51e..84e95f7 100644
--- a/arch/arm/cpu/arm_cortexa8/ti81xx/sys_info.c
+++ b/arch/arm/cpu/arm_cortexa8/ti81xx/sys_info.c
@@ -123,7 +123,7 @@ u32 pg_val_ti816x(u32 pg1_val, u32 pg2_val)
 u32 pg_val_ti814x(u32 pg1_val, u32 pg2_val)
 {
 	/* PG2.1 devices should read 0x3 as chip rev */
-	if (PG2_1 == get_cpu_rev())
+	if (PG2_1 <= get_cpu_rev())
 		return pg2_val;
 	else
 		return pg1_val;

